### ISSUE TITLE
problem in skating_analysis vignette:

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/hyper2#18]
Problem reported by Kurt by gmail, 5 May 2019:

```
checking re-building of vignette outputs ... [30s/54s] WARNING
Error(s) in re-building vignettes:
  ...
--- re-building ‘icons.Rmd’ using rmarkdown
--- finished re-building ‘icons.Rmd’

--- re-building ‘skating_analysis.Rmd’ using rmarkdown
Quitting from lines 98-125 (skating_analysis.Rmd) 
Error: processing vignette 'skating_analysis.Rmd' failed with diagnostics:
initial value in ‘vmmin’ is not finite
--- failed re-building ‘skating_analysis.Rmd’

--- re-building ‘table_tennis_serve.Rmd’ using rmarkdown
--- finished re-building ‘table_tennis_serve.Rmd’

--- re-building ‘hyper2.Rnw’ using Sweave
Loading required package: cubature
--- finished re-building ‘hyper2.Rnw’

--- re-building ‘integration.Rnw’ using Sweave
--- finished re-building ‘integration.Rnw’

SUMMARY: processing the following file failed:
  ‘skating_analysis.Rmd’

Error: Vignette re-building failed.
Execution halted
```

**Comment by RobinHankin** [comment]
He also said it in the email:

The Debian r-devel-gcc checks are currently performed using OpenBLAS
(version 0.3.5) instead of the default refblas: apparently your code is
making unrealistic assumptions about numerical precision/tolerances?


Please correct before 2019-05-18 to safely retain your package on CRAN.

**Comment by RobinHankin** [comment]
Kurt emailed me:

Robin,

Not sure how much I can help debug.

One, which version of OpenBLAS do you have on your Mac?  My Debian
system has 0.3.5.

Two, when I run the vignette, the problem is in

R> for(i in seq_len(nrow(problem_constraints))){
+ jj <- maxp(skating, startp=start, give=TRUE,fcm=problem_constraints[i,], fcv=0)
+    fullout[[i]] <- jj
+    out[i] <- jj$value
+ }
Error in optim(theta.old, fun, gradient, control = control, method = method,  : 
  initial value in ‘vmmin’ is not finite
Calls: maxp -> constrOptim -> optim

I tried the effect of doing the maxp() call for all possible i, and
found that it fails for i = 1, and works for the others.  Any ideas?

Best
-k

**Comment by RobinHankin** [comment]
Can now reproduce the bug by following Peter Langfelder's advice on R-devel mailing list, and replacing libRblas.dylib with the appropriate openblas dylib file.  (which was ```lib/R/lib/libRblas.dylib```).  I need to do a bit more work on this because it is still not clear what the origin of the problem is although increasing the tolerance in ```maxp()``` from ```1e-6``` to ```1e-5``` stops the error

**Comment by RobinHankin** [comment]
Can also reproduce the bug  on linux by making openblas 0.3.7 and then  ```cp libopenblas.so lib/R/lib/libRblas.so```.  File ```libopenblas.so``` is made by the ```Makefile``` in openblas.

**Comment by RobinHankin** [comment]
I want a minimal working example so I can report this as a bug, but this is proving difficult.

**Comment by RobinHankin** [comment]
Today I was trying to remember how I resolved this issue and got confused because the offending vignette, skating_analysis.Rmd, still appears under the vignettes/ directory.   But it is under the inst/ directory in the CRAN version of the package.   There is a difference between the CRAN version of the package and the version under github.  I just moved the file  to the inst/ directory, built the package with ```R CMD build```, and whacked it off to CRAN.

Transcript of email sent from me to Kurt on 16 May 2019:

Hi Kurt, thanks for your advice.  I managed to reproduce the error by using OpenBLAS, and am pretty certain that the problem does not lie in the hyper2 package: the error occurs in src/appl/optim.c  for no reason I can see, and everything works fine with the default BLAS on R-devel.  On the other hand, I've not isolated a minimum reproducible example either.  As I say, I'm not 100% that the problem isn't with hyper2.

Anyway, I've "solved" the problem by moving the vignette into the inst/ directory, and uploaded to CRAN just now.  Thanks for your help, 

best wishes

Robin

[not
